### PR TITLE
Feature/csv parser

### PR DESCRIPTION
**Pull request by BilalAlGhazi**
**Please make sure the following boxes are checked before submitting your Pull Request:**

- [X] I've added this PR's link to its Asana task(s)
- [X] If this PR adds new code that is not going to change in the near future, it includes unit tests to cover it.

## Adds parseCsv function to new-polymath-shared
Function wraps [Papa Parse](https://www.papaparse.com/) to provide validation functionality. Function accepts CSV data in a string or file upload control.
Code includes Validator class that provides general validation library. Pushed code includes unit test coverage.
web3 added as dependency to use the address validation functionality when validating addresses in CSV files.

### CODE REVIEW

File: packages/new-polymath-shared/src/utils/validator.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  };
+
+  private isString = (value: any) => {
+    if (typeof value === 'string') {
+      return true;
+    }
+    return false;
+  };
+
+  private isNotEmpty = (value: any) => {
```
Comment (RafaelVidaurre) [inline packages/new-polymath-shared/src/utils/validator.ts]: We can simplify some of these functions in the following way:

```typescript
const isNotEmpty = (value: any) => value !== '' && value !== null && typeof value !== 'undefined';
```

File: packages/new-polymath-shared/src/utils/__tests__/utils.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import { hashObj, delay } from '../index';
+import { hashObj, delay, parseCsv } from '../index';
+import { validRange } from 'semver';
```
Comment (RafaelVidaurre) [inline packages/new-polymath-shared/src/utils/__tests__/utils.ts]: Not sure why this was imported, buy I'm guessing your editor auto-imported this maybe? 😉

File: packages/new-polymath-shared/src/utils/validator.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import Web3 from 'web3';
+
+export class Validator {
+  private readonly web3: Web3 = new Web3();
```
[the remark this answers is not in the document]
Comment (BilalAlGhazi) [reply packages/new-polymath-shared/src/utils/validator.ts]: For some reason when I use the `Web3.Utils.isAddress` I get the following error. `@types/web3` is added as a dependency.

image

**Comment by pabloruiz55** [comment]
It’s possible web3 0.x is being used instead of web3 1.x.

https://github.com/ethereum/wiki/wiki/JavaScript-API

File: packages/new-polymath-shared/src/utils/csvParser.ts [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+          }
+          result.push(resultObj);
+          if (resultObj.isRowValid) {
+            validRows++;
+          } else {
+            errorRows++;
+          }
+        }
+      };
+      const complete = (results: ParseResult) => {
```
[the remark this answers is not in the document]
Comment (BilalAlGhazi) [reply packages/new-polymath-shared/src/utils/csvParser.ts]: The `complete` object is passed to papaparse through the `config` object.

**Pull request merged**